glslc/README.asciidoc

```diff
@@ -236,6 +236,66 @@ that all `warning:` messages are shown as `error:` and any warnings will cause
 a non-zero exit code from `glslc`. If `-w` is specified the warnings
 generated are suppressed before they are converted to errors.
 
+=== Dependency Generation Options
+
+==== `-M` or `-MM`
+
+`-M` generates *make* dependencies, outputs a rule suitable for *make*
+describing the dependencies of the main source file. Instead of outputting
+the result of preprocessing, the preprocessor outputs one *make* rule
+containing the SPIR-V object file name for that source file, a colon, and the
+names of all the included files.
+
+Unless specified explicitly (with `-MT`), the SPIR-V object file name in the
+generated *make* rules follows the rules of  <<output-file-naming,Output File
+Naming>> as in `-c` compilation stage.
+
+The dependency info file name can be specified by `-o` and `-MF` options. When
+both are specified, `-o` option is ignored.
+
+Passing `-M` implies `-E`, and suppresses warnings with an implicit `-w`.
+
+`-MM` is an alias of `-M`.
+
+E.g., `glslc -M main.vert` will dump `main.vert.spv: main.vert <other included
+files>` to stdout. `glslc -M main.vert -o dep_info` will generate a file
+`dep_info` to store the *make* rule of `main.vert`. `glslc -M main.vert -MT
+target` will have *make* rule with `target` as the object file name: `target:
+main.vert <other included files>`.
+
+==== `-MD`
+
+`-MD` generates *make* dependencies and compiles. Unlike `-M`, `-MD` does not
+implies `-E` and it outputs both compilation results and dependencies. If the
+user specifies an dependency info file name by `-MF`, the compiler will take
+that name, otherwise, the compiler will take the `-o` option if one is given,
+or deduce from the input file name if a `-o` option is not given then generate
+the name by the rules of <<output-file-naming, Output File Naming>> plus a
+*.d* suffix.
+
+E.g., `glslc -c -MD main.vert` will generate `main.vert.spv` as the SPIR-V
+object file and `main.vert.spv.d` as the dependency info file. `glslc -c -MD
+main.vert -o shader_obj` will generate `shader_obj` as the SPIR-V object file
+and `shader_obj.d` as the dependency info file. `glslc -c -MD main.vert -MF
+dep_info` will generate `main.vert.spv` as the SPIR-V object file and `dep_info`
+as the dependency info file.
+
+==== `-MF`
+
+`-MF` lets you specify the dependency info file name when used with `-M` or
+`-MD`.
+
```

Comment:
What happens if I have multiple source files, and -c, and use -MF.
Should be an error.